package NNeuralRankerProtocol;

message TEmbedding {
    optional string Name = 1;
    optional uint32 Len = 2;
};

message TSlice {
    optional string Name = 1;
    repeated uint32 Indices = 2;
};

message TInputPart {
    oneof Vector {
        TEmbedding Embedding = 1;
        TSlice Slice = 2;
    }
};

message TInput {
    optional string Name = 1;
    repeated TInputPart InputParts = 2;
};

message TMetaInfo {
    repeated TInput Inputs = 1;
    optional string Fingerprint = 2;
    optional string WorkflowId = 3;
    optional string Date = 4;
    repeated string InputNodeNames = 5;
    optional string OutputNodeName = 6;
    optional uint32 OutputVectorLen = 7;
};

message TRankModel {
    optional bytes GraphSerialized = 1;
    optional TMetaInfo MetaInfo = 2;
};
